src/OpenCensus.Abstractions/Trace/ISampler.cs

```diff
@@ -35,7 +35,11 @@ public interface ISampler
         /// <param name="hasRemoteParent">Indicates whether it was a remote parent.</param>
         /// <param name="traceId">Trace ID of a span to be created.</param>
         /// <param name="spanId">Span ID of a span to be created.</param>
-        /// <param name="name">Name of a span to be created.</param>
+        /// <param name="name"> Name of a span to be created. Note, that the name of the span is settable.
+        /// So this name can be changed later and <see cref="ISampler"/> implementation should assume that.
+        /// Typical example of a name change is when <see cref="ISpan"/> representing incoming http request
+        /// has a name of url path and then being updated with route name when rouing complete.
```

Comment:
why do we do this through span name and not through the `http.route` attribute?

Reply:
span name should have low cardinality. Thus we set route into name. Perhaps we also need to put route into the attribute.